src/components/ModalEmailSubscription.vue

```diff
@@ -0,0 +1,99 @@
+<script setup lang="ts">
+defineProps<{
+  open: boolean;
+}>();
+
+type ModalView = 'SUBSCRIBE' | 'SUCCESS';
+
+const emit = defineEmits(['close']);
+const { web3Account } = useWeb3();
+const email = ref('');
```

Comment:
We should keep refs and composables separate

Reply:
what do you mean ?